src/operations/change_trust.js

```diff
@@ -2,24 +2,30 @@ import isUndefined from 'lodash/isUndefined';
 import { Hyper } from 'js-xdr';
 import BigNumber from 'bignumber.js';
 import xdr from '../generated/stellar-xdr_generated';
+import { ChangeTrustAsset } from '../change_trust_asset';
 
 const MAX_INT64 = '9223372036854775807';
 
 /**
  * Returns an XDR ChangeTrustOp. A "change trust" operation adds, removes, or updates a
  * trust line for a given asset from the source account to another. The issuer being
- * trusted and the asset code are in the given Asset object.
+ * trusted and the asset code are in the given ChangeTrustAsset object.
  * @function
  * @alias Operation.changeTrust
  * @param {object} opts Options object
- * @param {Asset} opts.asset - The asset for the trust line.
+ * @param {ChangeTrustAsset} opts.asset - The change trust asset for the trust line.
  * @param {string} [opts.limit] - The limit for the asset, defaults to max int64.
  *                                If the limit is set to "0" it deletes the trustline.
  * @param {string} [opts.source] - The source account (defaults to transaction source).
  * @returns {xdr.ChangeTrustOp} Change Trust operation
  */
 export function changeTrust(opts) {
   const attributes = {};
+
+  if (!(opts.asset instanceof ChangeTrustAsset)) {
+    throw new TypeError('options.asset must be a ChangeTrustAsset');
+  }
```

Comment:
Since this error will happen in the XDR anyway I'm actually okay without introducing this check here. We don't do this in other places so it's better to be consistent.

Reply:
I've decided to let it there but I'm open to discussing it in https://github.com/stellar/js-stellar-base/pull/454#discussion_r695798768.